src/huggingface_hub/repocard_data.py

```diff
@@ -514,23 +527,22 @@ def eval_results_to_model_index(
     # Here, we make a map of those pairs and the associated EvalResults.
     task_and_ds_types_map = defaultdict(list)
     for eval_result in eval_results:
-        task_and_ds_pair = (eval_result.task_type, eval_result.dataset_type)
-        task_and_ds_types_map[task_and_ds_pair].append(eval_result)
+        task_and_ds_types_map[eval_result.unique_identifier].append(eval_result)
 
     # Use the map from above to generate the model index data.
     model_index_data = []
-    for (task_type, dataset_type), results in task_and_ds_types_map.items():
+    for eval_result_identifier, results in task_and_ds_types_map.items():
         data = {
             "task": {
-                "type": task_type,
+                "type": eval_result_identifier[0],
                 "name": results[0].task_name,
             },
             "dataset": {
                 "name": results[0].dataset_name,
-                "type": dataset_type,
-                "config": results[0].dataset_config,
-                "split": results[0].dataset_split,
-                "revision": results[0].dataset_revision,
+                "type": eval_result_identifier[1],
+                "config": eval_result_identifier[2],
+                "split": eval_result_identifier[3],
+                "revision": eval_result_identifier[4],
```

Comment:
@lewtun Small change but here I would take all information from the first item of `results`. I think it's clearer which metadata we are using (instead of the `eval_result_identifier[...]`). And imagine in the future we return a hash instead of a tuple (for example if `dataset_config` has to be hashed), we wouldn't have to change this snippet. What do you think ?

```suggestion
    for results in task_and_ds_types_map.values():
        # All items from `results` share same metadata
        sample_result = results[0]
        data = {
            "task": {
                "type": sample_result.task_type,
                "name": sample_result.task_name,
            },
            "dataset": {
                "name": sample_result.dataset_name,
                "type": sample_result.dataset_type,
                "config": sample_result.dataset_config,
                "split": sample_result.dataset_split,
                "revision": sample_result.dataset_revision,
```

Reply:
Good idea! Added in [b2bb6d6](https://github.com/huggingface/huggingface_hub/pull/1214/commits/b2bb6d6cc3391977bda02e17073166791a418db2) because GitHub wouldn't let me commit your suggestion in the UI

Reply:
Nice ! CI is green so feel free to merge :) Thanks again for fixing this thing !